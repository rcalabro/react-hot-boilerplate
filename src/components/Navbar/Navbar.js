import React, { PropTypes, Component } from 'react'
import { IndexLink } from 'react-router'
import Nav from 'react-bootstrap/lib/Nav'
import NavItem from 'react-bootstrap/lib/NavItem'
import BSNavbar from 'react-bootstrap/lib/Navbar'
import Image from 'react-bootstrap/lib/Image'
import { IndexLinkContainer, LinkContainer } from 'react-router-bootstrap'

class Navbar extends Component {

  handleLogout = (event) => {
    event.preventDefault();
    this.props.signout();
  };

  render() {
    const { user } = this.props;

    return (

      <BSNavbar fluid fixedTop>
        <BSNavbar.Header>
          <IndexLink to={'/'}>
            <BSNavbar.Brand>
                <Image src='/static/logo.svg'/>
            </BSNavbar.Brand>
          </IndexLink>
          <BSNavbar.Toggle />
        </BSNavbar.Header>
        <BSNavbar.Collapse>
          {!user &&
            <Nav pullRight>
              <LinkContainer to={'/signin'}>
                <NavItem eventKey={99}>Sign In</NavItem>
              </LinkContainer>
            </Nav>
          }
          {user &&
            <Nav>
                <LinkContainer to={'/checkout'}>
                  <NavItem eventKey={1}>Ads Checkout</NavItem>
                </LinkContainer>
                <LinkContainer to="/logout">
                  <NavItem eventKey={2} onClick={this.handleLogout}>Logout</NavItem>
                </LinkContainer>
            </Nav>
          }
          {user &&
            <p className='navbar-text pull-right'>Welcome, <strong>{user.name}</strong>.</p>
          }
        </BSNavbar.Collapse>
      </BSNavbar>
    )
  }
}

Navbar.propTypes = {
  user: PropTypes.object,
  signout: PropTypes.func.isRequired
}

export default Navbar
