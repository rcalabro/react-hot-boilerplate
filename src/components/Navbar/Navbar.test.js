import React from 'react';
import renderer from 'react-test-renderer'
import Navbar from './Navbar';

jest.mock('react-dom')

test('Navbar renders correctly', () => {
  const tree = renderer.create(
    <Navbar />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
