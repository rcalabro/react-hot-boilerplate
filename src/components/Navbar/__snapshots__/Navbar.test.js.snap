exports[`test Navbar renders correctly 1`] = `
<nav
  className="navbar navbar-default navbar-fixed-top">
  <div
    className="container-fluid">
    <div
      className="navbar-header">
      <img
        className="navbar-brand"
        src="/static/logo.svg" />
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button">
        <span
          className="sr-only">
          Toggle navigation
        </span>
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse">
      <ul
        className="nav navbar-nav navbar-right"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            action="push"
            href="#"
            onClick={[Function]}
            role="button">
            Home
          </a>
        </li>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            action="push"
            href="#"
            onClick={[Function]}
            role="button">
            Sign In
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
